### PR TITLE
#4739 - show link on preview if doc exists (story 3657)

### PR DESCRIPTION
**Pull request by stuart-bass-bjss**

### CODE REVIEW

File: app/pages/preview/createPreviewPageContext.js [hunk 1]
```diff
@@ -1,7 +1,16 @@
 export const createPreviewPageContext = ({ previewData }) => {
-  const context = {
+  let context = {
```
Comment (kat1906) [inline app/pages/preview/createPreviewPageContext.js]: this can be const

File: app/pages/preview/createPreviewPageContext.js [hunk 2]
```diff
@@ -1,7 +1,16 @@
...
     sections: previewData.sections,
   };
 
+  for(let key in context.sections)
+  { 
+    let section = context.sections[key];
```
Comment (kat1906) [inline app/pages/preview/createPreviewPageContext.js]: const

File: app/pages/preview/createPreviewPageContext.js [hunk 3]
```diff
@@ -1,7 +1,16 @@
...
+    if(section.answers && section.answers['document-name'])
+    {
```
Comment (alibombaye) [inline app/pages/preview/createPreviewPageContext.js]: surprised eslint didn't complain. I know this is a .NET convention but we have start on the brace on the same line

Comment (kat1906) [reply app/pages/preview/createPreviewPageContext.js]: I think it has failed in the pipeline

**Comment by kat1906** [comment]
Can you run the lint and the tests again as both are failing for me. `npm t` and `npm run lint` - we have a pre push hook to run the linting so I'm surprised it wasn't picked up! 

Also the createPreviewPageContext function would be better written using a different type of loop - I'm going to take a look now.

Also might be better to download VS code for writing JS as it picks up all these as you write it :)

File: app/pages/preview/createPreviewPageContext.test.js [hunk 4]
```diff
@@ -23,4 +23,32 @@ describe('createPreviewPageContext', () => {
 
     expect(context).toEqual(expectedContext);
   });
+
```
Comment (alibombaye) [inline app/pages/preview/createPreviewPageContext.test.js]: Another test for when the document name doesn't exist and then perhaps another if there is more than just one?

**Pull request merged**